lib/cfg-lexer.c

```diff
@@ -329,7 +306,7 @@ cfg_lexer_init_include_level_buffer(CfgLexer *self, CfgIncludeLevel *level,
                                     gsize length)
 {
   level->include_type = CFGI_BUFFER;
-  level->name = g_strdup(name);
+  level->lloc.name = g_intern_string(name);
```

Comment:
I don't fully understand this patch.

Why do we use intern strings? It populates a global hash table. Do we have some kind of a lifetime problem here?